### PR TITLE
EVG-6454 use githashes from patch modules

### PR DESCRIPTION
**Pull request by ybrill**
Manifests' records of module githashes are updated when a patch is created that includes module patches where the base commit is not the same as the one recorded in the manifest. This causes the manifest to be incorrect for other versions sharing the same manifest (the repotracker version as well as other patches off the same base commit).
This PR removes the manifest update in favor of dynamically adding a `ModuleOverrides` field to the manifest based on the patch requesting the manifest.

### CODE REVIEW

File: model/manifest/manifest.go [hunk 1]
```diff
@@ -13,11 +9,12 @@ const Collection = "manifest"
...
-	Revision    string             `json:"revision" bson:"revision"`
-	ProjectName string             `json:"project" bson:"project"`
-	Branch      string             `json:"branch" bson:"branch"`
-	Modules     map[string]*Module `json:"modules" bson:"modules"`
+	Id              string             `json:"id" bson:"_id"`
+	Revision        string             `json:"revision" bson:"revision"`
+	ProjectName     string             `json:"project" bson:"project"`
+	Branch          string             `json:"branch" bson:"branch"`
+	Modules         map[string]*Module `json:"modules" bson:"modules"`
+	ModuleOverrides map[string]string  `json:"module_overrides,omitempty" bson:"module_overrides,omitempty"`
```
Comment (tychoish) [inline model/manifest/manifest.go]: are we ever actually putting this in the database?

Comment (ybrill) [reply model/manifest/manifest.go]: no.
I removed the extra tag.

File: model/manifest/manifest.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ type Manifest struct {
 	ProjectName     string             `json:"project" bson:"project"`
 	Branch          string             `json:"branch" bson:"branch"`
 	Modules         map[string]*Module `json:"modules" bson:"modules"`
-	ModuleOverrides map[string]string  `json:"module_overrides,omitempty" bson:"module_overrides,omitempty"`
+	ModuleOverrides map[string]string  `json:"module_overrides,omitempty" bson:"omitempty"`
```
Comment (tychoish) [inline model/manifest/manifest.go]: you want this to be `bson:"-"`

Comment (ybrill) [reply model/manifest/manifest.go]: done.

**Pull request merged**